jax/experimental/compilation_cache/file_system_cache.py

```diff
@@ -37,6 +39,36 @@ def put(self, key: str, value: bytes):
     """Adds new cache entry, possibly evicting older entries."""
     if not key:
       raise ValueError("key cannot be empty")
-      #TODO(colemanliyah):implement eviction
-    with open(os.path.join(self._path, key), "wb") as file:
+
+    self._evict_entries_if_necessary(key, value)
+
+    self._create_a_file(self._path, key, value)
+
+  def _evict_entries_if_necessary(self, key:str, value:bytes):
+
+    directory_size = self._get_cache_directory_size()
+    new_file_size = len(value)
+
+    while new_file_size + directory_size > self._max_cache_size_bytes:
+      last_time = 0
+      file_to_delete = ""
+
+      for file_name in os.listdir(self._path):
+        file_to_inspect = os.path.join(self._path, file_name)
+        mtime = os.path.getmtime(file_to_inspect)
+        if mtime > last_time:
+          last_time = mtime
+          file_to_delete = file_to_inspect
+
+      os.remove(file_to_delete)
+      directory_size = self._get_cache_directory_size()
+
+  def _create_a_file(self, directory: str, key: str, value: bytes):
+    """Creates a new file, returns the size of the new file"""
+    path_to_new_file = os.path.join(directory, key)
+    with open(path_to_new_file, "wb") as file:
       file.write(value)
+
+  def _get_cache_directory_size(self):
+    """Retrieves the current size of the directory, self.path"""
+    return sum(os.path.getsize(file_in_directory) for file_in_directory in os.scandir(self._path) if file_in_directory.is_file())
```

Comment:
This line is very long. You can just use `file` or even `f` for `file_in_directory`. A general principle is that variable names can be shorter if they're only used in a very small amount of code, since it's easy to see where the variable is coming from so the name is less important to understanding what it is. You might also put a line break before the `if`.